api/provider/aws/deploy_delete.go

```diff
@@ -0,0 +1,52 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
+)
+
+func (d *DeployProvider) Delete(deployID string) error {
+	fqDeployID := addLayer0Prefix(d.Config.Instance(), deployID)
```

Comment:
the deploy id doesn't map directly to a family name. You'll need to do the task definition arn lookup here and pass that along to delete.